protocol/v2/ssv/validator/committee_queue.go

```diff
@@ -72,11 +72,33 @@ func (v *Committee) HandleMessage(logger *zap.Logger, msg *queue.DecodedSSVMessa
 
 // ConsumeQueue consumes messages from the queue.Queue of the controller
 // it checks for current state
-func (v *Committee) ConsumeQueue(logger *zap.Logger, slot phase0.Slot, handler MessageHandler) error {
-	ctx, cancel := context.WithCancel(v.ctx)
-	defer cancel()
-
+func (v *Committee) ConsumeQueue(
+	ctx context.Context,
+	logger *zap.Logger,
+	slot phase0.Slot,
+	handler MessageHandler,
+) error {
 	var q queueContainer
+
+	// in case of any error try to call the ctx.cancel to prevent the ctx leak
+	defer func() {
+		var ok bool
+
+		v.mtx.Lock()
+		q, ok = v.Queues[slot]
+		v.mtx.Unlock()
+
```

Comment:
`StartDuty` is already locking, so this is potential "deadlock" or just double locking, although this happens in a goroutine and the lock is released in start duty. Still I think its better to just pass the Queue `q` from `StartDuty` as we're already using the mutex to get it there.

by passing it to this function you don't need to use the mutex here at all.